src/helpers/localStorage.ts

```diff
@@ -9,17 +13,58 @@ export const LOCAL_STORAGE_KEYS = {
 
 export type LocalStorageKey = typeof LOCAL_STORAGE_KEYS[keyof typeof LOCAL_STORAGE_KEYS]
 
-function set<T>(key: LocalStorageKey, value: T): void {
-  localStorage.setItem(key, JSON.stringify(value))
+export const LOCAL_STORAGE_TYPES = {
+  'autorun-simulation': t.boolean,
+  'lang': t.string,
+  'log-scale': t.boolean,
+  'show-humanized-results': t.boolean,
+  'suppress-disclaimer': t.type({
+    version: t.number,
+    suppressShowAgain: t.boolean,
+  }),
+  'skip-landing-page': t.union([t.literal('true'), t.literal('false')]),
+}
+
+export type LocalStorageTypes = {
+  [K in LocalStorageKey]: t.TypeOf<typeof LOCAL_STORAGE_TYPES[K]>
 }
 
-function get<T>(key: LocalStorageKey): T | null {
+function set<K extends LocalStorageKey>(key: K, value: LocalStorageTypes[K]): void {
+  const type: t.Mixed = LOCAL_STORAGE_TYPES[key]
+  const encoded = type.encode(value)
+  localStorage.setItem(key, JSON.stringify(encoded))
+}
+
+function get<K extends LocalStorageKey>(key: K): LocalStorageTypes[K] | null {
   const value = localStorage.getItem(key)
+  if (value === null) {
+    return null
+  }
+
+  let obj: unknown
   try {
-    return value !== null ? JSON.parse(value) : value
+    obj = JSON.parse(value)
   } catch {
-    return (value as unknown) as T
```

Comment:
Previously we returned the value if parsing failed. I changed this to emit a warning and return `null` instead which I think is better behaviour!